lib/url/analyzers/index-of.js

```diff
@@ -13,18 +13,17 @@ function getBaseUrls(location) {
   const parsed = new url.URL(location)
   parsed.search = ''
 
-  let baseUrl = parsed.toString()
-  const filename = path.basename(baseUrl)
+  const baseUrl = parsed.toString()
 
-  // If the url ends with a filename, let’s remove it
-  if (filename.indexOf('.') > -1) {
-    baseUrl = path.dirname(baseUrl)
+  return {
+    baseUrl: normalize(baseUrl, {
+      stripWWW: false
+    }),
+    rootUrl: normalize(baseUrl, {
+      stripWWW: false,
+      removeDirectoryIndex: true
```

Comment:
this may need to be a configuration option (we can pass an array of regex) by default this remove all `^index\..*$` but  we may need to allow for more (stuff like `download.php` for example)